File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.project import ProjectMode, ProjectStatus


class ProjectCreate(BaseModel):
    client_name: str
    project_name: str
    jd_raw: str | None = None
    requirement_profile: dict | None = None
    mode: ProjectMode = ProjectMode.PRECISE


class ProjectUpdate(BaseModel):
    client_name: str | None = None
    project_name: str | None = None
    jd_raw: str | None = None
    requirement_profile: dict | None = None
    mode: ProjectMode | None = None
    status: ProjectStatus | None = None


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    client_name: str
    project_name: str
    jd_raw: str | None
    requirement_profile: dict | None
    mode: ProjectMode
    status: ProjectStatus
    folder_path: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/project.ts
export type ProjectStatus = 'active' | 'completed' | 'archived'

export interface ProjectCreate {
  client_name: string
  project_name: string
  jd_raw?: string | null
  requirement_profile?: Record<string, unknown> | null
}

export interface ProjectUpdate {
  client_name?: string | null
  project_name?: string | null
  jd_raw?: string | null
  requirement_profile?: Record<string, unknown> | null
  status?: ProjectStatus | null
}

export interface ProjectResponse {
  id: string
  client_name: string
  project_name: string
  jd_raw: string | null
  requirement_profile: Record<string, unknown> | null
  status: ProjectStatus
  folder_path: string | null
  created_at: string
  updated_at: string
}
